mlrun/runtimes/function.py

```diff
@@ -878,6 +880,15 @@ def invoke(
                                      see this link for more information:
                                      https://requests.readthedocs.io/en/latest/api/#requests.request
         """
+
+        # here we check that if default http trigger is disabled, function contains a custom http trigger
+        # Otherwise, the function is not invokable, so we raise an error
+        if "spec.triggers.http" not in self.spec.config and self.spec.disable_default_http_trigger:
+            raise mlrun.errors.MLRunPreconditionFailedError(
+                "Default http trigger creation is disabled and there is no any other custom http trigger, "
+                "so function can not be invoked via http. Either enable default http trigger creation or create"
+                "custom http trigger")
+
```

Comment:
I think this can be changed to inspect the function status and see if it does not have either external/internal invocation url - then not invokable. that as long as path does not include "://" of course.

Reply:
@liranbg this logic already exists in this function below, so should I delete this part then? As an option I can also add this check after 
`path = self._resolve_invocation_url(path, force_external_address)` in case of `not path`. wdyt?

Reply:
sounds good. just make sure you cover it with a test